server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -112,7 +115,25 @@ private static XContentBuilder mappings() {
         }
     }
 
-    public void getSynonymsSet(String resourceName, int from, int size, ActionListener<SynonymsSetResult> listener) {
+    public void getSynonymsSets(int from, int size, ActionListener<PagedResult<SynonymSetSummary>> listener) {
+        client.prepareSearch(SYNONYMS_ALIAS_NAME)
+            .setSize(0)
+            .addAggregation(new TermsAggregationBuilder(SYNONYM_SETS_AGG_NAME).field(SYNONYMS_SET_FIELD).order(BucketOrder.key(true)))
```

Comment:
We need to add `size` to `TermsAggregationBuilder`, as it will not return more than 10 results otherwise.